persist/fs/write.go

```diff
@@ -408,12 +428,27 @@ func (w *writer) writeIndexFileContents(
 			return fmt.Errorf("encountered duplicate ID: %s", id)
 		}
 
+		var encodedTags []byte
+		if tags := w.indexEntries[i].tags; tags != nil {
+			tagsIter.Reset(tags)
+			tagsEncoder.Reset()
+			if err := tagsEncoder.Encode(tagsIter); err != nil {
+				return err
+			}
+			data, ok := tagsEncoder.Data()
+			if !ok {
+				return errWriterEncodeTagsDataNotAccessible
```

Comment:
When does this happen

Reply:
Only if you haven't actually encoded anything yet, it should never occur in practice.